pandas/core/indexes/base.py

```diff
@@ -2190,19 +2190,24 @@ def __or__(self, other):
     def __xor__(self, other):
         return self.symmetric_difference(other)
 
-    def _get_consensus_name(self, other):
+    def _get_setop_name(self, other):
+        # GH 9943 9862
         """
-        Given 2 indexes, give a consensus name meaning
-        we take the not None one, or None if the names differ.
-        Return a new object if we are resetting the name
+        Given 2 indexes, give the name appropriate for set
+        operations meaning we take the name if both names
+        are the same, or None if the names differ.
         """
-        if self.name != other.name:
-            if self.name is None or other.name is None:
-                name = self.name or other.name
-            else:
-                name = None
-            if self.name != name:
-                return self._shallow_copy(name=name)
+        return self.name if self.name == other.name else None
+
+    def _get_setop_name_object(self, other):
+        """
+        Given 2 indexes, give a setop name and object, meaning
+        we use _get_setop_name() to return the name, and then
+        return a new object if we are resetting the name
+        """
+        name = self._get_setop_name(other)
+        if self.name != name:
```

Comment:
I think might be able to dispense with this entirely, and just *always* do

```
from pandas.core.ops import get_op_result_name

....

return self._shallow_copy(name=get_op_result_name(self, other))
```

Reply:
do this in ``_wrap_setup_result``

Reply:
So if I eliminate `self.get_setop_name_object` and use `_wrap_setop_result`, other tests fail, because those tests assume the corner cases return the same object.